### ISSUE TITLE
Boolean column with all nulls causes `npm run sources` sources to fail

### ISSUE THREAD

**Issue by archiewood** [evidence-dev/evidence#1504]
## Steps To Reproduce

Create a source query with a boolean column containing only nulls

```sql
select null::bool as country, 100 as sales 
union all 
select null::bool as country, 200 as sales
union all 
select null::bool as country, 300 as sales
```

### Environment

- Database: postgres, duckdb (assume also others but not tested)

## Expected Behavior

A column with nulls should be returned

## Actual Behaviour

```
Invalid argument error: Need at least 1 bytes for bitmap in buffers[0] in array of type Boolean, but got 0
```

![image](https://github.com/evidence-dev/evidence/assets/58074498/821f43ef-ec97-497c-bd46-79b622bfbb7e)

## Workarounds

Don't select this column with nulls as it is contains no information anyway

**Comment by mcrascal** [comment]
Only occurs on bool columns

**Comment by csjh** [comment]
Awaiting apache/arrow#39776

**Comment by mcrascal** [comment]
@csjh in the meantime can we provide something more informative in the error message?

**Comment by csjh** [comment]
Yeah, but something that just occurred to me is that we're still pinned on version 11.0.0, so even if it is fixed upstream we'll still have the faulty version (at least until duckdb-wasm bumps their versions?). Not sure what the best message to put in the warning would be considering that.

**Comment by csjh** [comment]
How bad would it be to just convert boolean columns to true/false? Or maybe having a special case when the column is fully null?

**Comment by archiewood** [comment]
Special case when column is fully null feels not too bad.

I'd probably advocate for one of the following
1. Error Message along the lines of: 
  - `Error: Column "column_name" is of type Boolean and contains only null values, which is not allowed. Consider removing the column from the query or casting it to another type`
2. Cast it to a string by default, and print a warning: 
  - `Warning: Column "column_name" (type Boolean) contains only null values so has been cast to string`
3. Fill with `false` by default, and print a warning: 
  - `Warning: Column "column_name" (type Boolean) contains only null values so has been infilled with `false` values.`

**Comment by archiewood** [comment]
@csjh is this done?

**Comment by csjh** [comment]
Fixed by #1521
